### PR TITLE
logging feature: fprintf replaced by rcutils_log

### PR DESCRIPTION
**Pull request by yuraSomatic**

### CODE REVIEW

File: kdl_parser/CMakeLists.txt [hunk 1]
```diff
@@ -6,6 +6,7 @@ find_package(ament_cmake_ros REQUIRED)
 find_package(orocos_kdl REQUIRED)
 find_package(urdf REQUIRED)
 find_package(urdfdom_headers REQUIRED)
+find_package(rcutils)
```
Comment (clalancette) [inline kdl_parser/CMakeLists.txt:9]:
```suggestion
find_package(rcutils REQUIRED)
```

Also, please put this in alphabetical order in the list above.

File: kdl_parser/CMakeLists.txt [hunk 2]
```diff
@@ -23,7 +24,8 @@ target_include_directories(${PROJECT_NAME}
 ament_target_dependencies(${PROJECT_NAME}
   orocos_kdl
   urdf
-  urdfdom_headers)
+  urdfdom_headers
+  rcutils)
```
Comment (clalancette) [inline kdl_parser/CMakeLists.txt:28]: Alphabetic order, please.

File: kdl_parser/src/kdl_parser.cpp [hunk 3]
```diff
@@ -86,9 +88,10 @@ KDL::Joint toKdl(urdf::JointSharedPtr jnt)
         return KDL::Joint(jnt->name, F_parent_jnt.p, F_parent_jnt.M * axis, KDL::Joint::TransAxis);
       }
     default: {
-        fprintf(
-          stderr, "Converting unknown joint type of joint '%s' into a fixed joint\n",
-          jnt->name.c_str());
+        rcutils_log(nullptr,
```
Comment (clalancette) [inline kdl_parser/src/kdl_parser.cpp:91]: Can we instead use:

`RCUTILS_LOG_WARN_NAMED` here?

(same goes for all of the other uses below)

**Review by clalancette — CHANGES_REQUESTED** [review]
I've left a few review comments.  Also, this is failing CI because of some uncrustify and cpplint errors.  Please fix those as well.

**Comment by SammyRamone** [comment]
Is there any progress on this? Currently we get the following blob of spam each time we start our software due to the issue that is addressed in this PR.
The proposed changes seem to make sense. As @yuraSomatic did not further responde, could you (@clalancette) just do these changes and merge the PR? 

```
[robot_state_publisher-3] Link torso had 6 children
[robot_state_publisher-3] Link neck had 1 children
[robot_state_publisher-3] Link head had 3 children
[robot_state_publisher-3] Link camera had 0 children
[robot_state_publisher-3] Link camera_optical_frame had 0 children
[robot_state_publisher-3] Link imu_frame_2 had 0 children
[robot_state_publisher-3] Link l_hip_1 had 1 children
[robot_state_publisher-3] Link l_hip_2 had 1 children
[robot_state_publisher-3] Link l_upper_leg had 1 children
[robot_state_publisher-3] Link l_lower_leg had 1 children
[robot_state_publisher-3] Link l_ankle had 1 children
[robot_state_publisher-3] Link l_foot had 6 children
[robot_state_publisher-3] Link llb had 1 children
[robot_state_publisher-3] Link l_cleat_l_back had 0 children
[robot_state_publisher-3] Link llf had 1 children
[robot_state_publisher-3] Link l_cleat_l_front had 0 children
[robot_state_publisher-3] Link lrb had 1 children
[robot_state_publisher-3] Link l_cleat_r_back had 0 children
[robot_state_publisher-3] Link lrf had 1 children
[robot_state_publisher-3] Link l_cleat_r_front had 0 children
[robot_state_publisher-3] Link l_sole had 0 children
[robot_state_publisher-3] Link l_toe had 0 children
[robot_state_publisher-3] Link l_shoulder had 1 children
[robot_state_publisher-3] Link l_upper_arm had 1 children
[robot_state_publisher-3] Link l_lower_arm had 1 children
[robot_state_publisher-3] Link l_wrist had 0 children
[robot_state_publisher-3] Link r_hip_1 had 1 children
[robot_state_publisher-3] Link r_hip_2 had 1 children
[robot_state_publisher-3] Link r_upper_leg had 1 children
[robot_state_publisher-3] Link r_lower_leg had 1 children
[robot_state_publisher-3] Link r_ankle had 1 children
[robot_state_publisher-3] Link r_foot had 6 children
[robot_state_publisher-3] Link rlb had 1 children
[robot_state_publisher-3] Link r_cleat_l_back had 0 children
[robot_state_publisher-3] Link rlf had 1 children
[robot_state_publisher-3] Link r_cleat_l_front had 0 children
[robot_state_publisher-3] Link rrb had 1 children
[robot_state_publisher-3] Link r_cleat_r_back had 0 children
[robot_state_publisher-3] Link rrf had 1 children
[robot_state_publisher-3] Link r_cleat_r_front had 0 children
[robot_state_publisher-3] Link r_sole had 0 children
[robot_state_publisher-3] Link r_toe had 0 children
[robot_state_publisher-3] Link r_shoulder had 1 children
[robot_state_publisher-3] Link r_upper_arm had 1 children
[robot_state_publisher-3] Link r_lower_arm had 1 children
[robot_state_publisher-3] Link r_wrist had 0 children
[robot_state_publisher-3] Link imu_frame had 0 children
```

**Comment by clalancette** [comment]
> The proposed changes seem to make sense. As @yuraSomatic did not further responde, could you (@clalancette) just do these changes and merge the PR?

There's a practical problem with doing that, in that I don't have permissions to push to @yuraSomatic's branch.  But I can open a new PR that has these changes.

That said, by default this is still going to print out those messages to the console.  The good news, though, is that you'll have more options to control what is being printed.

**Pull request closed without merging**